packages/framework/src/types/config.types.ts

```diff
@@ -23,4 +23,10 @@ export type ClientOptions = {
    * Defaults to true.
    */
   strictAuthentication?: boolean;
+
+  /**
+   * Whether to compile controls.
+   * Defaults to true.
+   */
+  compileControls?: boolean;
```

Comment:
Currently, this is a global option. However, we could make it more granular in the future, for example, configurable on a per-step basis.